contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java

```diff
@@ -187,8 +189,8 @@ public int next() {
     }
 
     int rowCount = 0;
-    done:
-    for (; rowCount < TARGET_RECORD_COUNT; rowCount++) {
+    // if first row is larger than allowed max size in batch, it will be added anyway
+    do {
```

Comment:
Still need to monitor row count: row count cannot exceed 64K. So, loop termination is either exceeds memory limit OR reaches max row count. For row count, might as well use the original limit, unless we know enough to pick a better limit.